### ISSUE TITLE
Unwarp resample leading to out of bounds error

### ISSUE THREAD

**Issue by mgxd** [poldracklab/sdcflows#318]
```
Node: fmriprep_0_0_wf.single_subject_voice969_wf.func_preproc_ses_1_task_emosent_run_01_wf.unwarp_wf.resample
Working directory: /scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/unwarp_wf/resample

Node inputs:

in_coeff = ['/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/coeff2epi_wf/map_coeff/reoriented_base_fieldcoef_fixed_space-target.nii.gz']
in_data = ['/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0000.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0001.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0002.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0003.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0004.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0005.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0006.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0007.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0008.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0009.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0010.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0011.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0012.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0013.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0014.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0015.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0016.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0017.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0018.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0019.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0020.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0021.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0022.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0023.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0024.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0025.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0026.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0027.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0028.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0029.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0030.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0031.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0032.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0033.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0034.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0035.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0036.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0037.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0038.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0039.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0040.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0041.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0042.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0043.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0044.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0045.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0046.nii.gz', '/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_split/vol0047.nii.gz']
in_xfms = ['/scratch/fmriprep_0_0_wf/single_subject_voice969_wf/func_preproc_ses_1_task_emosent_run_01_wf/bold_hmc_wf/fsl2itk/mat2itk.txt']
num_threads = 3
pe_dir = ['j-']
ro_time = [0.0738312]

Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node
    result["result"] = node.run(updatehash=updatehash)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 527, in run
    result = self._run_interface(execute=True)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 645, in _run_interface
    return self._run_command(execute)
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 771, in _run_command
    raise NodeExecutionError(msg)
nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node resample.

Traceback:
	concurrent.futures.process._RemoteTraceback: 
	"""
	Traceback (most recent call last):
	  File "/opt/conda/lib/python3.9/concurrent/futures/process.py", line 243, in _process_worker
	    r = call_item.fn(*call_item.args, **call_item.kwargs)
	  File "/opt/conda/lib/python3.9/concurrent/futures/process.py", line 202, in _process_chunk
	    return [fn(*args) for args in chunk]
	  File "/opt/conda/lib/python3.9/concurrent/futures/process.py", line 202, in <listcomp>
	    return [fn(*args) for args in chunk]
	  File "/opt/conda/lib/python3.9/site-packages/sdcflows/interfaces/bspline.py", line 624, in _b0_resampler
	    if unwarp.fit(distorted_img):
	  File "/opt/conda/lib/python3.9/site-packages/sdcflows/transform.py", line 99, in fit
	    wmat = grid_bspline_weights(spatialimage, moved_cs)
	  File "/opt/conda/lib/python3.9/site-packages/sdcflows/transform.py", line 415, in grid_bspline_weights
	    colloc_ax = BSpline.design_matrix(x, t, 3)[:, 2:-2]
	  File "/opt/conda/lib/python3.9/site-packages/scipy/interpolate/_bsplines.py", line 416, in design_matrix
	    raise ValueError(f'Out of bounds w/ x = {x}.')
	ValueError: Out of bounds w/ x = [-2.83567261 -2.28938216 -1.7430917  -1.19680125 -0.6505108  -0.10422035
	  0.4420701   0.98836055  1.534651    2.08094145  2.6272319   3.17352235
	  3.7198128   4.26610325  4.8123937   5.35868416  5.90497461  6.45126506
	  6.99755551  7.54384596  8.09013641  8.63642686  9.18271731  9.72900776
	 10.27529821 10.82158866 11.36787911 11.91416957 12.46046002 13.00675047
	 13.55304092 14.09933137 14.64562182 15.19191227 15.73820272 16.28449317
	 16.83078362 17.37707407 17.92336452 18.46965497 19.01594543 19.56223588
	 20.10852633 20.65481678 21.20110723 21.74739768 22.29368813 22.83997858
	 23.38626903 23.93255948 24.47884993 25.02514038 25.57143083 26.11772129
	 26.66401174 27.21030219 27.75659264 28.30288309 28.84917354 29.39546399
	 29.94175444 30.48804489 31.03433534 31.58062579 32.12691624].
	"""

	The above exception was the direct cause of the following exception:

	Traceback (most recent call last):
	  File "/opt/conda/lib/python3.9/site-packages/nipype/interfaces/base/core.py", line 398, in run
	    runtime = self._run_interface(runtime)
	  File "/opt/conda/lib/python3.9/site-packages/sdcflows/interfaces/bspline.py", line 373, in _run_interface
	    ) = zip(*outputs)
	  File "/opt/conda/lib/python3.9/concurrent/futures/process.py", line 559, in _chain_from_iterable_of_lists
	    for element in iterable:
	  File "/opt/conda/lib/python3.9/concurrent/futures/_base.py", line 609, in result_iterator
	    yield fs.pop().result()
	  File "/opt/conda/lib/python3.9/concurrent/futures/_base.py", line 439, in result
	    return self.__get_result()
	  File "/opt/conda/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
	    raise self._exception
	ValueError: Out of bounds w/ x = [-2.83567261 -2.28938216 -1.7430917  -1.19680125 -0.6505108  -0.10422035
	  0.4420701   0.98836055  1.534651    2.08094145  2.6272319   3.17352235
	  3.7198128   4.26610325  4.8123937   5.35868416  5.90497461  6.45126506
	  6.99755551  7.54384596  8.09013641  8.63642686  9.18271731  9.72900776
	 10.27529821 10.82158866 11.36787911 11.91416957 12.46046002 13.00675047
	 13.55304092 14.09933137 14.64562182 15.19191227 15.73820272 16.28449317
	 16.83078362 17.37707407 17.92336452 18.46965497 19.01594543 19.56223588
	 20.10852633 20.65481678 21.20110723 21.74739768 22.29368813 22.83997858
	 23.38626903 23.93255948 24.47884993 25.02514038 25.57143083 26.11772129
	 26.66401174 27.21030219 27.75659264 28.30288309 28.84917354 29.39546399
	 29.94175444 30.48804489 31.03433534 31.58062579 32.12691624].
```

**Comment by effigies** [comment]
This is basically a bounding box error, affected by `BSpline.design_matrix()` which wants additional knots to handle the target resampling points. Proposed fix incoming.

**Comment by effigies** [comment]
Fixed by #319.
